src/requirements.h

```diff
@@ -190,6 +210,8 @@ struct requirement_data {
         const requirement_data disassembly_requirements() const;
 
     private:
+        std::string id_;
```

Comment:
> It was initially but I refactored to plain string when I realised why would an instance need to refer to itself?

The id shouldn't care where it's stored, the `string_id` does not only provide access to the underlying type, but also provides a hint of what type of id it is. It also provide accidental misuse, like comparing an quality-id with an item-id.

But let's go with your argument. Why does the class expose that id via the getter - if it does not need to refer to itself?

The seemingly random switch from `string_id` to `std::string` is at least confusing.